core/src/main/java/org/apache/accumulo/core/client/admin/TableSummary.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.accumulo.core.client.admin;
+
+import java.util.Collections;
+import java.util.Map;
+
+public class TableSummary {
+  public enum Status {
+    /**
+     * Requested summary was computed for all data
+     */
+    PRESENT,
+
+    /**
+     * Requested summary were computed for some data.
+     */
+    PARTIALLY_PRESENT,
```

Comment:
I was only thinking of persisted data... I think summary info could be tracked for in memory maps though.  Would have to specify in interface that it should be safe to call summarize() multiple times.